texture-font.c

```diff
@@ -736,14 +787,22 @@ texture_font_load_glyph( texture_font_t * self,
 	// half a pixel each to get crisp rendering
 	glyph->s0       = x - 0.5;
 	glyph->t0       = y - 0.5;
-	glyph->s1       = x + glyph->width - 0.5;
-	glyph->t1       = y + glyph->height - 0.5;
+	glyph->s1       = x + tgt_w - 0.5;
+	glyph->t1       = y + tgt_h - 0.5;
     }
     // Discard hinting to get advance
-    FT_Load_Glyph( self->face, glyph_index, FT_LOAD_RENDER | FT_LOAD_NO_HINTING);
+    FT_Load_Glyph( self->face, glyph_index,
+		   ((self->atlas->depth == 4) ? FT_LOAD_COLOR : 0) |
+		   FT_LOAD_RENDER | FT_LOAD_NO_HINTING);
     slot = self->face->glyph;
-    glyph->advance_x = slot->advance.x / HRESf;
-    glyph->advance_y = slot->advance.y / HRESf;
+    if( self->atlas->depth == 4 ) {
+	// color fonts use actual pixels, not subpixels
+	glyph->advance_x = slot->advance.x * self->scale;
```

Comment:
This part seems to be messing up my normal text rendering - I'm using an atlas with depth 4 so I can store ordinary text and emojis in the same atlas, and thus render in the same pass. If I use the calculation in the "else" block, then rendering is fine.